[AArch64] Adjust comparison constant if adjusting it means less instructions

AZero13 opened the pull request:
Prefer constants that require less instructions to materialize, in both Global-ISel and Selection-DAG

davemgreen commented on llvm/lib/Target/AArch64/GISel/AArch64PostLegalizerLowering.cpp:
```diff
@@ -656,14 +656,13 @@ tryAdjustICmpImmAndPred(Register RHS, CmpInst::Predicate P,
 
-  if (!IsMaterializableInSingleInstruction(OriginalC) &&
-      IsMaterializableInSingleInstruction(C))
+  if (NumberOfInstrToLoadImm(OriginalC) > NumberOfInstrToLoadImm(C))
```
Do you have any tests where gisel changes? The difference between 2 and 3?

AZero13 replied:
It's to match seldag

davemgreen replied:
Yep - do we have any test cases for it though, where we could either generate a 2-cost or 3-cost imm, and we now pick the 2-cost version?

AZero13 committed:
[AArch64] Adjust comparison constant if adjusting it means less instructions

Prefer constants that require less instructions to materialize, in both Global-ISel and Selection-DAG

AZero13 committed:
ult

AZero13 committed:
Fix

AZero13 commented:
@davemgreen Fixed!

AZero13 committed:
Update AArch64PostLegalizerLowering.cpp

davemgreen reviewed (approved):
Thanks. LGTM.
